modules/local/xml_vcf/resources/usr/bin/shortvariant.py

```diff
@@ -63,18 +65,101 @@ def extract_data_from_short_variant(short_variant):
+            if record.seq[start:end].upper()==Seq(del_sequence).upper().reverse_complement():
+                updated=start-1
+                short_variant['REF']=record.seq[updated:end].upper()
+                short_variant['ALT']=record.seq[updated:end-len(del_sequence)].upper()
+            else:
+                print("ErrorD2: Expected sequence does not match reverse complement of provided sequence: Original-%s vs Provided-%s" % (record.seq[start:end].upper(),Seq(del_sequence).upper().reverse_complement()))
+        elif strand==1:
+            ### If strand is forward, sanity check by finding matching sequence
+            if record.seq[start:end].upper()==Seq(del_sequence).upper():
+                updated=start-1
+                short_variant['REF']=record.seq[updated:end].upper()
+                short_variant['ALT']=record.seq[updated:end-len(del_sequence)].upper()
+            else:
+                print("ErrorD2: Expected sequence does not match provided sequence: Original-%s vs Provided-%s" % (record.seq[start:end].upper(),Seq(del_sequence).upper()))
+        else:
+            print("ErrorD4: RETURN STRAND DOES NOT COMPUTE")
+
+    elif "ins" in short_variant.get('cds-effect'):
+        ###Populate initial variables
+        start=int(short_variant.get("POS"))
+        added_sequence=re.findall("[A-Z]+",short_variant.get("cds-effect"))[0]
+        end=start+1
+        gene=short_variant.get('gene').replace("MLL2","KMT2D")
+
+        ###Query for strand information from API
+        url='https://www.genenetwork.nl/api/v1/gene/%s' % gene.lower()
+        response=requests.get(url)
+
+        if response.status_code!=200:
+            print("ERRORI1: Unable to ping https://www.genenetwork.nl/api/v1/gene/%s" % gene.lower())
+            exit(1)
+            
+        strand=response.json()['gene']['strand']
+
+        if strand==-1:
+            updated=start
+            short_variant['REF']=record.seq[updated:end].upper()
+            short_variant['ALT']=added_sequence+record.seq[updated:end].upper()
+        elif strand==1:
+            updated=start
+            short_variant['REF']=record.seq[updated:end].upper()
+            short_variant['ALT']=record.seq[updated:end].upper()+added_sequence
+        else:
+            print("ERRORI2: UNKNOWN STRAND FOUND IN %s" % (gene) )
+            exit(1)
+    elif ">" in short_variant.get('cds-effect'):
+
+        short_variant['REF']=re.findall("[a-zA-Z]+",short_variant.get('cds-effect'))[0]
+        short_variant['ALT']=re.findall("[a-zA-Z]+",short_variant.get('cds-effect'))[1]
+        ### Label single or multi based on length of affected REF sequence
+        short_variant['variant-type']= 'multiple-nucleotide-substitution' if len(re.findall("[a-zA-Z]+",short_variant.get('cds-effect'))[0])>1 else 'single-nucleotide-substitution'
+    else:
+        print("ERROR: UNKNOWN VARIANT TYPE DETECTED" )
+
+
 def split_MNV_SNV(df):
```

Comment:
When split MNV to SNV, there are some INVALID variants (REF==ALT) being created, we need to exclude those.

Reply:
need to filter redundant variants and will keep duplicate values